libs/sdk-ui-dashboard/src/model/commandHandlers/layout/moveSectionItemHandler.ts

```diff
@@ -199,6 +202,54 @@ function validateAndResolve(commandCtx: MoveSectionItemContext) {
     }
 }
 
+function getCommonPath(fromItemPath: ILayoutItemPath, toItemPath: ILayoutItemPath) {
+    const emptyPath: ILayoutItemPath = [];
+    const shorterPath = fromItemPath.length < toItemPath.length ? fromItemPath : toItemPath;
+    return shorterPath.reduce((acc, item, index) => {
+        if (areSameCoordinates(fromItemPath[index], toItemPath[index])) {
+            acc.push(item);
+        }
+        return acc;
+    }, emptyPath);
+}
+
+function toIsBeforeFrom(toItemPath: ILayoutItemPath, fromItemPath: ILayoutItemPath) {
+    return (
+        getSectionIndex(toItemPath) <= getSectionIndex(fromItemPath) ||
+        (getSectionIndex(toItemPath) === getSectionIndex(fromItemPath) &&
+            getItemIndex(toItemPath) <= getItemIndex(fromItemPath))
+    );
+}
+
+function requiresItemShift(fromItemPath: ILayoutItemPath, toItemPath: ILayoutItemPath) {
+    const commonPath = fromItemPath.slice(0, toItemPath.length);
+    return (
+        hasSamePredPath(commonPath.slice(0, -1), toItemPath.slice(0, -1)) &&
+        toIsBeforeFrom(toItemPath, commonPath)
+    );
+}
+
+function getSectionPathWithItemsShifted(
```

Comment:
These methods could be moved to `libs/sdk-ui-dashboard/src/_staging/layout/coordinates.ts`? Will we reuse them? One way or another, please use AI to generate some unit tests. It would be nice to freeze the logic if someone would touch it later to be sure it still works as it should.

Reply:
Will add UTs and move it. So far it seems that most of the functions is not reusable